### PR TITLE
Issue 42 swagger spec

### PR DESCRIPTION
**Pull request by Lull3rSkat3r**
Adds a minimal swagger spec to be automatically generated in `notebook-http` mode.

### CODE REVIEW

**Comment by Lull3rSkat3r** [comment]
Waiting for PR #65 to be merged before rebasing.

File: kernel_gateway/services/cell/parser.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import re
```
Comment (parente) [inline kernel_gateway/services/cell/parser.py]: Jupyter copyright.

File: kernel_gateway/services/swagger/builders.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from ..cell.parser import APICellParser
```
Comment (parente) [inline kernel_gateway/services/swagger/builders.py]: Jupyter copyright.

File: kernel_gateway/services/swagger/builders.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from os import path
+
+class SwaggerSpecBuilder(object):
```
Comment (parente) [inline kernel_gateway/services/swagger/builders.py]: A class-level doc string would be good explaining what the class does.

File: kernel_gateway/services/swagger/handlers.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+import tornado.web
+from tornado.log import access_log
+from tornado import gen
+import json
+from .builders import *
+from random import random
+
+class SwaggerSpecHandler(tornado.web.RequestHandler):
```
Comment (parente) [inline kernel_gateway/services/swagger/handlers.py]: Docstring at class level at least.

File: kernel_gateway/tests/services/cell/test_parser.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from kernel_gateway.services.cell.parser import *
+import unittest
+import re
+class TestAPICellParserUtils(unittest.TestCase):
```
Comment (parente) [inline kernel_gateway/tests/services/cell/test_parser.py]: We never had tests for this code before? The PR reviewer is slipping ... :sleeping:

File: kernel_gateway/services/cell/parser.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import sys
+
+class EndpointSortStrategies(object):
```
Comment (parente) [inline kernel_gateway/services/cell/parser.py]: Pythonic way of doing this would be just to put the function in the top-level module namespace.

**Comment by Lull3rSkat3r** [comment]
Updated as per comments above and included some more test cases.

**Pull request merged**